packages/server/src/calendar/calendar.controller.ts

```diff
@@ -51,32 +59,92 @@ export class CalendarController {
     }
   }
 
-  @Get(':cid')
-  async getAllEvents(@Param('cid') cid: number): Promise<CalendarModel[]> {
-    const events = await CalendarModel.find({
-      where: {
-        course: cid,
-      },
+  @Patch(':id')
+  async updateEvent(
+    @Param('id') id: string,
+    @Body() body: Partial<Calendar>,
+  ): Promise<CalendarModel> {
+    const event = await CalendarModel.findOne(id);
+
+    if (!event) {
+      console.error('Event not found with ID: ' + id);
+      throw new HttpException('Event not found', HttpStatus.NOT_FOUND);
+    }
+    Object.keys(body).forEach((key) => {
+      event[key] = body[key];
     });
-    if (events === null || events === undefined) {
-      console.error(ERROR_MESSAGES.courseController.courseNotFound + 'events ');
+
+    try {
+      await event.save();
+      return event;
+    } catch (err) {
+      console.error(err);
       throw new HttpException(
-        ERROR_MESSAGES.courseController.courseNotFound,
-        HttpStatus.NOT_FOUND,
+        'Calendar update error',
+        HttpStatus.INTERNAL_SERVER_ERROR,
       );
     }
+  }
 
-    return events;
+  @Get(':cid')
+  async getAllEvents(
+    @Param('cid', ParseIntPipe) cid: number,
+  ): Promise<CalendarModel[]> {
+    const events = await CalendarModel.find({ where: { course: cid } });
+    return events || [];
+  }
+
+  @Get(':cid/:date')
+  async getEventsForDay(
+    @Param('cid') cid: number,
+    @Param('date') date: string,
+    @Query('timezone') timezone: string,
+  ): Promise<CalendarModel[]> {
+    // Parse the date string into a Date object
+    const targetDate = new Date(date);
+
+    // Get the start and end of the day
+    const startOfDay = new Date(targetDate);
+    startOfDay.setHours(0, 0, 0, 0);
+    const endOfDay = new Date(targetDate);
+    endOfDay.setHours(23, 59, 59, 999);
+    // Determine the day of week index from the target date
+    const dayOfWeek = targetDate.getDay().toString();
+    // Retrieve all events for the given course
+    const events = await CalendarModel.find({
+      where: { course: cid },
+    });
+    console.log('all events', events);
+    // Filter to get events occurring on the target date
+    const filteredEvents = events.filter((event) => {
+      if (!event.daysOfWeek) {
+        // For one-time events, check if they occur on the target date
+        return (
+          new Date(event.start) >= startOfDay && new Date(event.end) <= endOfDay
+        );
+      } else {
+        // For recurring events, check if the target day is a match and within the event's date range
+        return (
+          event.daysOfWeek.includes(dayOfWeek) &&
+          new Date(event.start) <= targetDate &&
+          (!event.endDate || new Date(event.endDate) >= targetDate)
+        );
+      }
+    });
+    console.log(filteredEvents);
+    return filteredEvents;
   }
 
   @Delete(':id/delete')
-  async deleteCalendarEvent(
-    @Param('id') eventId: number,
+  async deleteQuestionType(
```

Comment:
Endpoint is not sage guarded by course/user role

Reply:
Gonna need to pass in a courseId/cid and then use guards

```ts
@UseGuards(CourseRolesGuard)
  @Roles(Role.TA, Role.PROFESSOR)
```

ALSO: the id param MUST be changed to like calId or something, since CourseRolesGuard will see the id param and assume its a course id and then not good